File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

//prototypes
int count_words(string t);
int sentences(string t);
int letters(string t);

int main(void)
{
    string text = get_string("Text: ");
    int w = count_words(text);
    float s = (float)sentences(text);
    float l = (float)letters(text);
    float avgw = w / 100.0;

    //find average sentences and letters
    s /= avgw;
    l /= avgw;

    //determines grade level
    int index = round((0.0588 * l) - (0.296 * s) - 15.8);

    //prints grade level
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

//counts the total number of words
int count_words(string t)
{
    int spaces = 0;
    for (int i = 0; i < strlen(t); i++)
    {
        if (t[i] == ' ')
        {
            spaces++;
        }
    }
    return spaces + 1;
}
//counts the total number of letters
int letters(string t)
{
    int total = 0;
    for (int i = 0; i < strlen(t); i++)
    {
        if (isalpha(t[i]))
        {
            total++;
        }
    }
    return total;
}

//counts the total number of sentences
int sentences(string t)
{
    int total = 0;
    for (int i = 0; i < strlen(t); i++)
    {
        //checks for punctuation
        if (t[i] == '.' || t[i] == '!' || t[i] == '?')
        {
            total++;
        }
    }
    return total;
}

File: sentimental-readability/readability.py
# TODO
from cs50 import get_string

# prompts user for text input
text = get_string("Text: ")

# count our values
letters = 0
words = 1
sentences = 0
for i in text:
    if i == ' ':
        words += 1
    elif (i.lower() >= 'a' and i.lower() <= 'z'):
        letters += 1
    elif (i == '!' or i == '.' or i == '?'):
        sentences += 1

# determine the Coleman-Liau index
avgw = words/100
L = letters / avgw
print(L)
S = sentences / avgw
print(S)

index = round(0.0588 * L - 0.296 * S - 15.8)

# printing the corresponding grade level
if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")
